src/searchbar.cpp

```diff
@@ -177,19 +177,21 @@ void SearchBarLineEdit::fetchMoreSuggestions()
 
 void SearchBarLineEdit::onScroll(int value)
 {
+    /* m_completer.popup()->currentIndex() being invalid means the user used the
+       keyboard up/down to scroll past the end.
+    */
+    if (!m_completer.popup()->currentIndex().isValid())
+        return onScrollPastEnd();
+
     const auto suggestionScroller = m_completer.popup()->verticalScrollBar();
     const auto scrollMin = suggestionScroller->minimum();
     const auto scrollMax = suggestionScroller->maximum();
     const bool scrolledToEnd = value == suggestionScroller->maximum();
-
     if (m_aboutToScrollPastEnd)
     {
         if (scrolledToEnd)
-        {
             /* The user's intention to scroll past end has been confirmed */
-            fetchMoreSuggestions();
-            m_aboutToScrollPastEnd = false; /* Relax until next time */
-        }
+            onScrollPastEnd();
```

Comment:
Please restore the braces - those may be omitted only if the body is a single liner but comments count as part of a body!